lib/scryptenc/scryptenc.c

```diff
@@ -664,32 +719,63 @@ scryptdec_file(FILE * infile, FILE * outfile,
 	crypto_aes_key_free(key_enc_exp);
 
 	/* Did we exit the loop due to a read error? */
-	if (ferror(infile)) {
+	if (ferror(C->infile)) {
 		rc = 13;
-		goto err1;
+		goto err0;
 	}
 
 	/* Did we read enough data that we *might* have a valid signature? */
 	if (buflen < 32) {
 		rc = 7;
-		goto err1;
+		goto err0;
 	}
 
 	/* Verify signature. */
 	HMAC_SHA256_Final(hbuf, &hctx);
 	if (memcmp(hbuf, buf, 32)) {
 		rc = 7;
-		goto err1;
+		goto err0;
 	}
 
-	/* Zero sensitive data. */
-	insecure_memzero(dk, 64);
+	/* Success! */
+	return (0);
+
+err0:
+	/* Failure! */
+	return (rc);
+}
+
+/**
+ * scryptdec_file(infile, outfile, passwd, passwdlen,
+ *     maxmem, maxmemfrac, maxtime, verbose, force):
+ * Read a stream from infile and decrypt it, writing the resulting stream to
+ * outfile.  If ${force} is 1, do not check whether decryption
+ * will exceed the estimated available memory or time.
+ */
+int
+scryptdec_file(FILE * infile, FILE * outfile, const uint8_t * passwd,
+    size_t passwdlen, size_t maxmem, double maxmemfrac, double maxtime,
+    int verbose, int force)
+{
+	struct scryptdec_file_cookie * C;
+	int rc;
+
+	/* Check header, including passphrase. */
+	if ((rc = scryptdec_file_prep(infile, passwd, passwdlen, maxmem,
```

Comment:
I would write this as `if ((rc = ...) != 0)` since I think that's clearer (especially to people who have trouble counting parentheses).